modules/images/webp-uploads/load.php

```diff
@@ -683,3 +683,91 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
+ *
+ * Updates webp image when original image is edited
+ *
+ * @since n.e.x.t
+ *
+ * @param bool|null       $override  Value to return instead of saving. Default null.
+ * @param string          $file      Name of the file to be saved.
+ * @param WP_Image_Editor $image     The image editor instance.
+ * @param string          $mime_type The mime type of the image.
+ * @param int             $post_id   Attachment post ID.
+ *
+ * @return bool|null
+ */
+function webp_uploads_update_image_onchange( $override, $file, $image, $mime_type, $post_id ) {
+	if ( $override !== null ) {
+		return $override;
+	}
+
+	$valid_mime_transforms      = webp_uploads_get_upload_image_mime_transforms();
+	$original_directory         = pathinfo( $file, PATHINFO_DIRNAME );
+	$filename                   = pathinfo( $file, PATHINFO_FILENAME );
+	$sizes                      = get_intermediate_image_sizes();
+	$_wp_additional_image_sizes = wp_get_additional_image_sizes();
+	$target                     = ! empty( $_REQUEST['target'] ) ? preg_replace( '/[^a-z0-9_-]+/i', '', $_REQUEST['target'] ) : '';
+	$nocrop                     = false;
+	$current_mime_type          = get_post_mime_type( $post_id );
+	$allowed_mimes              = array_flip( wp_get_mime_types() );
+
+	foreach ( $valid_mime_transforms[ $mime_type ] as $targeted_mime ) {
+		if ( $targeted_mime === $current_mime_type ) {
+			continue;
+		}
+
+		if ( ! isset( $allowed_mimes[ $targeted_mime ] ) || ! is_string( $allowed_mimes[ $targeted_mime ] ) ) {
+			return new WP_Error( 'image_mime_type_invalid', __( 'The provided mime type is not allowed.', 'performance-lab' ) );
+		}
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $targeted_mime ) ) ) {
+			return new WP_Error( 'image_mime_type_not_supported', __( 'The provided mime type is not supported.', 'performance-lab' ) );
+		}
+
+		$extension   = explode( '|', $allowed_mimes[ $targeted_mime ] );
+		$destination = trailingslashit( $original_directory ) . "{$filename}.{$extension[0]}";
+
+		$image->save( $destination, $targeted_mime );
+
+		$new_image = wp_get_image_editor( $destination );
+
+		if ( is_wp_error( $new_image ) ) {
+			return $new_image;
+		}
+
+		if ( 'thumbnail' === $target ) {
+			$nocrop = true;
+		}
+
+		if ( ! isset( $sizes ) ) {
+		  continue;
+		}
+
+		$_sizes = array();
+
+		foreach ( $sizes as $size ) {
+
+			if ( isset( $_wp_additional_image_sizes[ $size ] ) ) {
+				$width  = (int) $_wp_additional_image_sizes[ $size ]['width'];
+				$height = (int) $_wp_additional_image_sizes[ $size ]['height'];
+				$crop   = ( $nocrop ) ? false : $_wp_additional_image_sizes[ $size ]['crop'];
+			} else {
+				$height = (int) get_option( "{$size}_size_h" );
+				$width  = (int) get_option( "{$size}_size_w" );
+				$crop   = ( $nocrop ) ? false : get_option( "{$size}_crop" );
+			}
+
+			$_sizes[ $size ] = array(
+				'width'  => $width,
+				'height' => $height,
+				'crop'   => $crop,
+			);
+		}
+
+		$new_image->multi_resize( $_sizes );
```

Comment:
I thought at first this won't work, but looking at the WP core code in more depth it indeed will. Even without passing the MIME type, the method will reuse the MIME type previously set, which will happen on editor instantiation _and_ when calling the `save()` method.

That is a great find @kirtangajjar! 🎉 

cc @adamsilverstein @mitogh We can probably simplify _a lot_ of our implementation so far by using this approach.